### PR TITLE
Information about CPU architecture for GitHub hosted runners

### LINKED ISSUE #13794: Document VM Architecture for GitHub Hosted Runners
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

### What part(s) of the article would you like to see updated?

At present, it doesn't seem that there's any mention of the architectures (for example, x86, x64, armv7, arm64) that are supported. Even if there's only one supported (I wouldn't be shocked if they're all the same!), it would be really helpful to have that documented for use cases where an end-user is running a build where architecture matters.

### Additional information

I'm looking to cross-reference the GitHub Hosted Actions Runners with a support matrix for a major open source project to consider migrating to Actions as our primary CI rather than a self-hosted Jenkins instance. We support a very large number of architectures, so knowing which we'd need to set up self-hosted VMs for would be extremely helpful.

### PR DESCRIPTION
**Pull request by thispsj**
### Why:

Closes #13794 

### What's being changed:

As mentioned by the member of GitHub Actions team, I've specified that Linux and Windows GitHub hosted runners have amd64 architecture and macOS runners have Intel x86_64 architecture.

Although the GitHub Actions member has specified that all runners are amd64 I've purposely written architecture as Intel x86_64 for macOS runners because macOS has only two archs : Intel x86_64 & arm64

### Check off the following:

- [X] I have reviewed my changes in staging.
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by NexusXe** [comment]
I feel that simply saying `2-core CPU (x86_64)` and `3-core CPU (x86_64)` would be better. Both "amd64" and "Intel x86_64" mean the same thing.

**Comment by janiceilene** [comment]
@thispsj Thanks so much for opening a PR! I'll get this triaged for review ⚡

**Comment by thispsj** [comment]
@janiceilene No worries!

**Comment by jaystyles** [comment]
@ dependably merge

**Comment by jaystyles** [comment]
@ dependabot merge

**Comment by jaystyles** [comment]
@ dependabot close

File: content/actions/using-github-hosted-runners/about-github-hosted-runners.md [hunk 1]
```diff
@@ -87,12 +87,12 @@ While the job runs, the logs and output can be viewed in the {% data variables.p
 ## Supported runners and hardware resources
 
 Hardware specification for Windows and Linux virtual machines:
-- 2-core CPU
+- 2-core CPU having `amd64` architecture.
```
Comment (martin389) [inline content/actions/using-github-hosted-runners/about-github-hosted-runners.md:90]:
```suggestion
- 2-core CPU (x86_64)
```

File: content/actions/using-github-hosted-runners/about-github-hosted-runners.md [hunk 2]
```diff
@@ -87,12 +87,12 @@ While the job runs, the logs and output can be viewed in the {% data variables.p
...
 - 7 GB of RAM
 - 14 GB of SSD space
 
 Hardware specification for macOS virtual machines:
-- 3-core CPU
+- 3-core CPU having `Intel x86_64` architecture.
```
Comment (martin389) [inline content/actions/using-github-hosted-runners/about-github-hosted-runners.md:95]:
```suggestion
- 3-core CPU (x86_64)
```

**Review by martin389 — APPROVED** [review]
- Added a couple of small edits.
- Got approval from GitHub SME 👍

**Pull request merged**